src/extensions/cli-extension.ts

```diff
@@ -30,60 +84,166 @@ async function getAuthToken(api: ApisauceInstance = disneyApi): Promise<any> {
+          mealPeriods.lunch
+        }`
       ),
       api.get(
-        `/api/wdpro/explorer-service/public/finder/dining-availability/80008297;entityType=destination?searchDate=${date}&partySize=2&mobile=false&mealPeriod=${mealPeriods.dinner}`
+        `/api/wdpro/explorer-service/public/finder/dining-availability/80008297;entityType=destination?searchDate=${date}&partySize=${partySize.toString()}&mobile=false&mealPeriod=${
+          mealPeriods.dinner
+        }`
       ),
     ]).then((res) => res.map((res) => (res.data as AvailabilityResponse)?.availability));
 
     const mergedResults: AvailabilityResponse = mergeAll(allResults);
 
     if (mergedResults) {
-      const hasOffers: DiningAvailability = Object.entries(mergedResults as AvailabilityResponse)
+      const hasOffers: DiningAvailabilities = Object.entries(mergedResults as AvailabilityResponse)
         .filter(([_id, location]) => location.availableTimes.some((time) => time.offers !== undefined))
-        .reduce((acc, [id, val]) => ((acc[id.split(';')[0]] = val), acc), {}); // use comma operator to return mutated acc
+        .reduce(
+          (acc, [key, val]) => (
+            (acc[key.split(';')[0]] = {
+              ...val,
+              card: mapping[key.split(';')[0]],
+              cleanedTimes: val.availableTimes.reduce((carry, time) => {
+                const offers = time.offers.map((offer) => ({
+                  dateTime: offer.dateTime,
+                  time: offer.time,
+                  //Direct Reservation Link
+                  directUrl: `https://disneyland.disney.go.com/dining-reservation/setup-order/table-service/?offerId[]=${offer.url}&offerOrigin=/dining/`,
+                }));
+                return [...carry, ...offers];
+              }, {}),
+            }),
+            acc
+          ),
+          {}
+        );
 
       const restaurantIds = Object.keys(hasOffers);
 
       if (restaurantIds.length === 0) {
-        console.log(`Didn't find anything. Checking again in 60s. ${numTries} total attempts.`);
+        print.warning(`No offers found for anything. Checking again in 60s. ${numTries} total attempts.`);
         setTimeout(() => {
-          checkTables(date, onSuccess, api, (numTries += 1), tables);
-        }, 60000);
-        // FIXME: Temporary, hard-coded search for Lamplight Lounge availability.
-      } else if (!restaurantIds.includes('19013078')) {
-        console.log(restaurantIds);
-        console.log(`Didn't find the restaurant we want. Checking again in 60s. ${numTries} total attempts.`);
-        setTimeout(() => {
-          checkTables(date, onSuccess, api, (numTries += 1), tables);
+          checkTables({ date, onSuccess, api, numTries: (numTries += 1), tables, print, ids });
         }, 60000);
       } else {
-        // Seems like we found something
-        onSuccess(hasOffers);
+        if (ids) {
+          await eachSeries(ids, async (id: string, callback) => {
```

Comment:
I am unfamiliar with the `async` package, but I wonder if we can achieve the desired behavior here without relying on a new package (esp. since you're just using the single operator).

I found this page on replacing the `async` lib with async/await, and it includes a specific example for `eachSeries`. 

https://github.com/jmjpro/async-package-to-async-await/blob/master/README.md

Reply:
Yeah you caught me being lazy (embarrassed). I really needed that reservation when I wrote this hahaha